darwin/importer/formats/nifti.py

```diff
@@ -0,0 +1,288 @@
+
+import darwin.datatypes as dt
+from darwin.importer.formats.nifti_schemas import nifti_import_schema
+from darwin.version import __version__
+
+
+def parse_path(path: Path) -> Optional[List[dt.AnnotationFile]]:
+    """
+    Parses the given ``nifti`` file and returns a ``List[dt.AnnotationFile]`` with the parsed
+    information.
+
+    Parameters
+    ----------
+    path : Path
+        The ``Path`` to the ``nifti`` file.
+
+    Returns
+    -------
+    Optional[List[dt.AnnotationFile]]
+        Returns ``None`` if the given file is not in ``json`` format, or ``List[dt.AnnotationFile]``
+        otherwise.
+    """
+    if not isinstance(path, Path):
+        path = Path(path)
+    if path.suffix != ".json":
+        return None
+    with open(path, "r") as f:
+        data = json.load(f)
+        validate(data, schema=nifti_import_schema)
+    nifti_annotations = data.get("data")
+    if nifti_annotations is None:
+        return None
+    annotation_files = []
+    for nifti_annotation in nifti_annotations:
+        annotation_file = _parse_nifti(
+            Path(nifti_annotation["label"]),
+            nifti_annotation["image"],
+            path,
+            class_map=nifti_annotation.get("class_map"),
+            mode=nifti_annotation.get("mode", "image"),
+        )
+        annotation_files.append(annotation_file)
+    return annotation_files
+
+
+def _parse_nifti(nifti_path: Path, filename: Path, json_path: Path, class_map: Dict, mode: str) -> dt.AnnotationFile:
+
+    img: np.ndarray = process_nifti(nib.load(nifti_path))
+
+    shape = img.shape
+    processed_class_map = process_class_map(class_map)
+    video_annotations = []
+    if mode == "instances":  # For each instance produce a video annotation
+        for class_name, class_idxs in processed_class_map.items():
+            if class_name == "background":
+                continue
+            class_img = np.isin(img, class_idxs).astype(np.uint8)
+            cc_img, num_labels = cc3d.connected_components(class_img, return_N=True)
+            for instance_id in range(1, num_labels):
+                video_annotation = get_video_annotation(cc_img, class_idxs=[instance_id], class_name=class_name)
+                if video_annotation:
+                    video_annotations.append(video_annotation)
+    elif mode == "image":  # For each frame and each class produce a single frame video annotation
+        for i in range(shape[-1]):
+            slice_mask = img[:, :, i].astype(np.uint8)
+            for class_name, class_idxs in processed_class_map.items():
+                frame_annotations = {}
+                if class_name == "background":
+                    continue
+                class_mask = np.isin(slice_mask, class_idxs).astype(np.uint8).copy()
+                polygon = mask_to_polygon(mask=class_mask, class_name=class_name)
+                if polygon is None:
+                    continue
+                frame_annotations[i] = polygon
```

Comment:
I'm a bit confused - we expect single frame sequence, and yet there's a variable here as index. Why is that?

Reply:
Hi Rafal, "image" mode is perhaps confusingly named. A volume has been uploaded but in "image" mode the annotations are per class and per frame. Explained in these two Looms https://www.loom.com/share/30833a63ef2c49bc8b8f8f0eadecb9fa and https://www.loom.com/share/b25929896c7e40c9836e8c48322ea5bc

Reply:
Ok, I think I get it now. This is not an image either way ;)